dss-api.yml

```diff
@@ -702,6 +702,118 @@ paths:
+        If bundle failed to be located, the error is immediately raised to the user.
+        Otherwise start copying the data in the background to the hot storage.
+        After some time period, the data will be removed.
+        TBD: This could be based on initial checkout time or based on last access time.
+      operationId: dss.api.bundles.checkout.post
+      parameters:
+        - name: uuid
+          in: path
+          description: A RFC4122-compliant ID for the bundle.
+          required: true
+          type: string
+          pattern: "[A-Za-z0-9]{8}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{12}"
+        - name: version
+          in: query
+          description: Timestamp of file creation in RFC3339.  If this is not provided, the latest version is returned.
+          required: false
+          type: string
+          format: date-time
+        - name: replica
+          in: query
+          description: Replica to fetch from.
+          required: true
+          type: string
+          enum: [aws, gcp, azure]
+        - name: json_request_body
+          in: body
+          required: true
+          schema:
+            type: object
+            properties:
+              email:
+                description: An email address to send status updates to.
+                type: string
+                format: email
+              destination:
+                description: User-owned destination storage bucket.
+                type: string
+            required:
+              - email
+      responses:
+        200:
+          description: Returned when the bundle UUID with optionally specified version exists and checkout has been initiated.
+          schema:
+            type: object
+            properties:
+              execution_id:
+                description: Handle of the checkout execution request.
+                type: string
+            required:
+              - execution_id
+        default:
+          description: Unexpected error
+          schema:
+            allOf:
+              - $ref: '#/definitions/Error'
+              - type: object
+                properties:
+                  code:
+                    type: string
+                    description: Machine-readable error code.  The types of return values should not be changed lightly.
+                    enum: [unhandled_exception, illegal_arguments, unknown_source_schema, file_already_exists, not_found]
+                required:
+                  - code
+
+  /bundles/checkout/{execution_id}:
+    get:
+      summary: Check status of the checkout request identified by execution id.
+      description: |
+        User makes a request to check the status of the checkout request made earlier. The request is identified
+        by execution ID that was returned to the user in response to the checkout request.
+      operationId: dss.api.bundles.checkout.get
+      parameters:
+        - name: execution_id
```

Comment:
`execution_id` is internal nomenclature that is not sufficiently specific for this purpose. Please use `checkout_job_id` or similar.